src/renderers/shared/fiber/ReactFiberErrorLogger.js

```diff
@@ -90,25 +71,29 @@ function logCapturedError(capturedError: CapturedError): void {
           `using the error boundary you provided, ${errorBoundaryName}.`;
       } else {
         errorBoundaryMessage =
-          `This error was initially handled by the error boundary ${errorBoundaryName}. ` +
+          `This error was initially handled by the error boundary ${errorBoundaryName}.\n` +
           `Recreating the tree from scratch failed so React will unmount the tree.`;
       }
     } else {
       errorBoundaryMessage =
-        'Consider adding an error boundary to your tree to customize error handling behavior. ' +
-        'See https://fb.me/react-error-boundaries for more information.';
+        'Consider adding an error boundary to your tree to customize error handling behavior.\n' +
+        'You can learn more about error boundaries at https://fb.me/react-error-boundaries.';
     }
-
-    console.error(
-      `${componentNameMessage} You should fix this error in your code. ${errorBoundaryMessage}\n\n` +
-        `${errorSummary}\n\n` +
-        `The error is located at: ${componentStack}\n\n` +
-        `The error was thrown at: ${formattedCallStack}`,
-    );
+    const combinedMessage =
+      `${componentNameMessage}\nYou can find its details in an earlier log.\n\n` +
+      `The component hierarchy was: ${componentStack}\n\n` +
+      `${errorBoundaryMessage}`;
+
+    // In development, we provide our own message with just the component stack.
+    // We don't include the original error message and JS stack because the browser
+    // has already printed it. Even if the application swallows the error, it is still
+    // displayed by the browser thanks to the DEV-only fake event trick in ReactErrorUtils.
+    console.error(combinedMessage);
   } else {
-    console.error(
-      `React caught an error thrown by one of your components.\n\n${error.stack}`,
-    );
+    // In production, we print the error directly.
+    // This will include the message, the JS stack, and anything the browser wants to show.
+    // We pass the error object instead of custom message so that the browser displays the error natively.
+    console.error(error);
```

Comment:
We should remove the `message` and `name` variables now. They don't seem to be used anymore.